Lib/test/test_math.py

```diff
@@ -2503,6 +2503,50 @@ def test_input_exceptions(self):
         self.assertRaises(TypeError, math.atan2, 1.0)
         self.assertRaises(TypeError, math.atan2, 1.0, 2.0, 3.0)
 
+    def test_exception_messages(self):
+        x = -1.1
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a nonnegative input, got {x}"):
+            math.sqrt(x)
+
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log(x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log(123, x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log2(x)
+        with self.assertRaises(ValueError,
+                               msg=f"expected a positive input, got {x}"):
+            math.log2(x)
+
+        x = decimal.Decimal(x)
```

Comment:
x is `Decimal('-1.100000000000000088817841970012523233890533447265625')`. Are you sure that you did not want to test with `Decimal('-1.1')`? Or maybe test `atanh()` with `Decimal('0.99999999999999999999')`?

Reply:
We just test that appropriate exception messages raised:)  Not that math_1() helper works correctly.